frontend/src/hooks/useIntervalRange.ts

```diff
@@ -0,0 +1,108 @@
+import getLocalStorageKey from 'api/browser/localstorage/get';
+import dayjs from 'dayjs';
+import getTimeString from 'lib/getTimeString';
+import { useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+import { getDiffs } from 'utils/getDiffs';
+
+import ROUTES from '../constants/routes';
+import {
+	getDefaultOption,
+	Time,
+} from '../container/TopNav/DateTimeSelection/config';
+
+interface UseIntervalRangeI {
+	getTime: () => [number, number] | undefined;
+	getCustomOrIntervalTime: (time: Time, currentRoute: string) => Time;
+}
+
+export function useIntervalRange(): UseIntervalRangeI {
+	const { search, pathname } = useLocation();
+	const params = new URLSearchParams(search);
+	const searchStartTime = params.get('startTime');
+	const searchEndTime = params.get('endTime');
+
+	const localstorageStartTime = getLocalStorageKey('startTime');
+	const localstorageEndTime = getLocalStorageKey('endTime');
+	const getQueryInterval = useCallback(
+		(searchStartTime: string) => {
+			const lastRefresh = dayjs(
+				new Date(parseInt(getTimeString(searchStartTime), 10)),
+			);
+			const { minutedDiff, hoursDiff, daysDiff } = getDiffs(lastRefresh);
+
+			if (daysDiff > 1) {
+				return '1week';
```

Comment:
if dayDiff is greater than 1 then we are returning 1 week don't you think this is scalable let's say in future we add more `Time` for that reason this logic will fail